pilot/pkg/xds/ecds.go

```diff
@@ -66,11 +66,9 @@ func (e *EcdsGenerator) Generate(proxy *model.Proxy, w *model.WatchedResource, r
 	}
 
 	secretResources := referencedSecrets(proxy, req.Push, w.ResourceNames)
+	updatedSecrets := model.ConfigsOfKind(req.ConfigsUpdated, kind.Secret)
 	// Check if the secret updates is relevant to Wasm image pull. If not relevant, skip pushing ECDS.
-	if !model.ConfigsHaveKind(req.ConfigsUpdated, kind.WasmPlugin) && !model.ConfigsHaveKind(req.ConfigsUpdated, kind.EnvoyFilter) &&
-		model.ConfigsHaveKind(req.ConfigsUpdated, kind.Secret) {
-		// Get the updated secrets
-		updatedSecrets := model.ConfigsOfKind(req.ConfigsUpdated, kind.Secret)
+	if len(updatedSecrets) > 0 {
```

Comment:
I think this isn't quite right. Imagine we have a push for 1 WASM and 1 secret. At L72 we set needsPush=false; we may never set it back to true, thus skipping the wasm

Reply:
I think ECDS earlier also was working like this 

```
if !model.ConfigsHaveKind(req.ConfigsUpdated, kind.WasmPlugin) && !model.ConfigsHaveKind(req.ConfigsUpdated, kind.EnvoyFilter) &&
		model.ConfigsHaveKind(req.ConfigsUpdated, kind.Secret) {
```
So it is checking if the configs updates does NOT have WASM and Envoy filter but have Secrets - then we push. This change is inline with this?

Reply:
This code can only make us NOT push. So you are making it so there are more cases when we would not push.

If we are pushing '1 WASM and 1 secret', this may incorrectly find "the 1 secret is not relevant" and skip pushing. This is wrong since the WASM part is relevant.

This all points to a testing gap